# Nettoyage des données V2

In [1]:
import s3fs
import pandas as pd

In [2]:
pd.set_option("display.max_columns", None)

### Récupération des données

In [3]:
from scripts.importation import process_all_years_s3

In [4]:
# Connexion au bucket S3
fs = s3fs.S3FileSystem(client_kwargs={"endpoint_url": "https://minio.lab.sspcloud.fr"})

ROOT_S3_PATH = "mvallat/diffusion/projet"

In [5]:
# Importation bucket S3
MY_BUCKET = "mvallat/diffusion"
FILE_PATH_S3 = f"{MY_BUCKET}/projet/data_brevets.parquet"

with fs.open(FILE_PATH_S3, 'rb') as file_in:
    data_brevets = pd.read_parquet(file_in)

data_brevets.head(10)

doc-number kind country  status publication_country publication_doc-number  \
0    3038352   A1      FR  PUBDEM                  FR                3038352   
1    3038278   A1      FR  PUBDEM                  FR                3038278   
2    3038409   A1      FR  PUBDEM                  FR                3038409   
3    3038425   A1      FR  PUBDEM                  FR                3038425   
4    3038316   A1      FR  PUBDEM                  FR                3038316   
5    3038399   A1      FR  PUBDEM                  FR                3038399   
6    3038381   A1      FR  PUBDEM                  FR                3038381   
7    3038328   A1      FR  PUBDEM                  FR                3038328   
8    3038251   A1      FR  PUBDEM                  FR                3038251   
9    3038374   A1      FR  PUBDEM                  FR                3038374   

  publication_date publication_bopinum publication_nature  \
0         20170106             2017-01             Brevet   
1         20170106             2017-01             Brevet   
2         20170106             2017-01             Brevet   
3         20170106             2017-01             Brevet   
4         20170106             2017-01             Brevet   
5         20170106             2017-01             Brevet   
6         20170106             2017-01             Brevet   
7         20170106             2017-01             Brevet   
8         20170106             2017-01             Brevet   
9         20170106             2017-01             Brevet   

                                     invention-title  \
0  ENSEMBLE POUR PANNEAU DE PORTE ET PANNEAU ASSOCIE   
1  VEHICULE HYBRIDE COMPRENANT UNE SUPPRESSION DE...   
2  PROCEDE DE CONCEPTION ASSISTEE PAR ORDINATEUR ...   
3  DOCUMENT ELECTRONIQUE TEL QU'UNE CARTE A PUCE ...   
4  PROCEDE DE SYNTHESE EN CONTINU D'UN ELASTOMERE...   
5  PROCEDE ET DISPOSITIF DE COMMANDE D'UNE CONSIG...   
6  DISPOSITIF CAPTEUR PREVU POUR ETRE EMBARQUE DE...   
7  VEHICULE FERROVIAIRE POUR LE NETTOYAGE DES FIL...   
8  PLANCHER PVC EN FIBRES COUPEES ET PROCEDE DE F...   
9  INSTALLATION THERMIQUE DE SECHAGE DE MATIERE P...   

                                 applicant_1_orgname  \
0                  FAURECIA INTERIEUR INDUSTRIE None   
1                        IFP ENERGIES NOUVELLES None   
2  CENTRE NATIONAL DE LA RECHERCHE SCIENTIFIQUE None   
3                         OBERTHUR TECHNOLOGIES None   
4  COMPAGNIE GENERALE DES ETABLISSEMENTS MICHELIN...   
5                                    SOFTATHOME None   
6                              BOGUSZEWSKI GUILLAUME   
7                                SNCF MOBILITES None   
8   ZHEJIANG KINGDOM PLASTICS INDUSTRY CO., LTD None   
9                               GROUPE H. LABBE None   

                               applicant_1_address-1       applicant_1_city  \
0                                     2, RUE HENNAPE               NANTERRE   
1                        1 ET 4 AVENUE DE BOIS-PREAU  RUEIL MALMAISON CEDEX   
2                                  3 RUE MICHEL ANGE                  PARIS   
3                         420 RUE D'ESTIENNE D'ORVES               COLOMBES   
4                                    12 COURS SABLON       CLERMONT-FERRAND   
5                           81 AVENUE FRANCOIS ARAGO               NANTERRE   
6         533, RUE DE JUPITER\nBATIMENT B- APPT. B21               JUVIGNAC   
7                                2 PLACE AUX ETOILES            SAINT-DENIS   
8  SOUTH ZONE, CHEZI BRIDGE, HESHAN TOWN,\nZHEJIA...         TONGXIANG CITY   
9           ZAC DE LA TERRE ROUGE RUE DE L'INDUSTRIE        TOURNAN EN BRIE   

  applicant_1_postcode applicant_1_country  \
0                92000                  FR   
1                92852                  FR   
2                75016                  FR   
3                92700                  FR   
4                63000                  FR   
5                92000                  FR   
6                34990 

### Observation des variables

#### **1) Type de brevet**

In [6]:
# Colonne "kind"
data_brevets["kind"].value_counts()

kind
A1    107749
A3      4441
NA       102
B1         3
Name: count, dtype: int64

Le document contient trois types de brevets :

- **A1** : demande de brevet  
- **A3** : demande de certificat d'utilité  
- **B1** : brevet délivré  


**Concernant les types B1 (n = 3)** : cette base recensant les demandes de brevet, elle n'est pas censée contenir un brevet B1.  
Probablement dû à une erreur de classification, nous pouvons donc retirer ces observations.

In [7]:
# Suppression des B1
data_brevets_clean = data_brevets[data_brevets["kind"] != "B1"]

# Vérification qu'on a supprimé 3 observations
data_brevets_clean.shape[0]


112292

**Concernant les types A3 (n = 4 441)** : les certificats d'utilité sont des certificats délivrés par l'INPI qui protègent l'innovation, tout comme les brevets. En revanche la durée de protection est raccourcie (10 ans contre 20 ans) et la procédure d'examen de la demande est allégée. Pour simplifier l'analyse en la rendant plus homogène, nous ne gardons pas ces demandes A3 dans notre base.

In [8]:
# Suppression des A3
data_brevets_clean = data_brevets_clean[data_brevets_clean["kind"] != "A3"]

# Vérification : suppression de 4 441 observations (112,292 - 4,441 = 107,851)
data_brevets_clean.shape[0]

107851

**Concernant les NA (n = 102)** : plus d'observations sont nécessaires pour décider de la manière de gérer ces NA.


In [9]:
# Observation des NA
brevets_na = data_brevets[data_brevets["kind"] == "NA"]
brevets_na


doc-number kind country status publication_country  \
285            NA   NA      NA     NA                None   
606            NA   NA      NA     NA                None   
831            NA   NA      NA     NA                None   
1153           NA   NA      NA     NA                None   
1543           NA   NA      NA     NA                None   
...           ...  ...     ...    ...                 ...   
28236          NA   NA      NA     NA                None   
28556          NA   NA      NA     NA                None   
28760          NA   NA      NA     NA                None   
29096          NA   NA      NA     NA                None   
107329    3145430   NA      FR    NEW                  FR   

       publication_doc-number publication_date publication_bopinum  \
285                      None             None                None   
606                      None             None                None   
831                      None             None                None   
1153                     None             None                None   
1543                     None             None                None   
...                       ...              ...                 ...   
28236                    None             None                None   
28556                    None             None                None   
28760                    None             None                None   
29096                    None             None                None   
107329                3145430         20240802             2024-31   

       publication_nature                                    invention-title  \
285                  None                                                 NA   
606                  None                                                 NA   
831                  None                                                 NA   
1153                 None                                                 NA   
1543                 None                                                 NA   
...                   ...                                                ...   
28236                None                                                 NA   
28556                None                                                 NA   
28760                None                                                 NA   
29096                None                                                 NA   
107329  Demande française  Surveillance de l’intégrité des données et con...   

       applicant_1_orgname     applicant_1_address-1 applicant_1_city  \
285                     NA                        NA               NA   
606                     NA                        NA               NA   
831                     NA                        NA               NA   
1153                    NA                        NA               NA   
1543                    NA                        NA               NA   
...                    ...                       ...              ...   
28236                   NA                        NA               NA   
28556                   NA                        NA               NA   
28760                   NA                        NA               NA   
29096                   NA                        NA               NA   
107329             AMADEUS  485 ROUTE DU PIN MONTARD             BIOT   

       applicant_1_postcode applicant_1_country applicant_2_orgname  \
285                      NA                  NA                  NA   
606                      NA                  NA                  NA   
831                      NA                  NA                  NA   
1153                     NA                  NA                  NA   
1543                     NA                  NA                  NA   
...                     ...                 ...                 ...   
28236                    NA                  NA                  NA   
28556                    NA

In [10]:
# Vérification du nombre d'informations réellement présentes
brevets_na["country"].nunique()
brevets_na["invention-title"].nunique()

2

On peut constater que seules deux observations contiennent des informations. Nous pouvons donc supprimer toutes les observations qui ne contiennent que des NA.

In [11]:
# Suppression des observations 100% NA
data_brevets_clean = data_brevets_clean[data_brevets_clean["kind"] != "NA"]

#### **2) Numéro du document**

In [12]:
# Vérification que doc_number est unique

nb_number = data_brevets_clean["doc-number"].nunique()
print("Nombre de doc-number uniques :", nb_number)

nb_number2 = data_brevets_clean["publication_doc-number"].nunique()
print("Nombre de publication_doc-number uniques :", nb_number2)

nb_observations = data_brevets_clean.shape[0]
print("Nombre total d'observations :", nb_observations)


Nombre de doc-number uniques : 107748
Nombre de publication_doc-number uniques : 107748
Nombre total d'observations : 107749


Nous pouvons constater qu'il y a une observation qui n'a pas un numéro unique.

In [13]:
doublons = data_brevets_clean[data_brevets_clean.duplicated(subset=["doc-number"], keep=False)]
print(doublons)

      doc-number kind country  status publication_country  \
12476    3050837   A1      FR  PUBDEM                  FR   
58919    3050837   A1      FR     NEW                  FR   

      publication_doc-number publication_date publication_bopinum  \
12476                3050837         20171103             2017-44   
58919                3050837         20171103             2017-44   

      publication_nature                                    invention-title  \
12476             Brevet  DISPOSITIF D'INFORMATION INERTIELLE, SAGITTALE...   
58919  Demande française  DISPOSITIF D'INFORMATION INERTIELLE, SAGITTALE...   

                  applicant_1_orgname      applicant_1_address-1  \
12476  BOARDING RING READING SAS None  661 CHE DU CHATEAU VALLON   
58919       BOARDING RING READING SAS  661 CHE DU CHATEAU VALLON   

      applicant_1_city applicant_1_postcode applicant_1_country  \
12476        OLLIOULES                83190                  FR   
58919        OLLIOULES         

#### **3) Statut**

La documentation de l'INPI nous apprend que la variable "status" est utilisée pour indiquer quel est l'évènement qui a entraîné la présence du brevet ou du CCP dans le fichier de la semaine. Les 

In [29]:
# Colonne "status"
data_brevets_clean["publication_nature"].value_counts()

publication_nature
Demande française    79625
Brevet               28124
Name: count, dtype: int64

In [14]:
data_brevets_clean.shape[0]

107749

#### **4) Abstract**
Les abstracts sont les résumés des brevets déposés. Il s'agit pour nous de vérifier qu'ils contiennent bien du texte d'une part, et qu'il n'y a pas de doublons d'autre part.

In [40]:
counts = data_brevets_clean["abstract"].value_counts()
print(counts[counts > 1])

abstract
NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             

On peut constater qu'il y a 1166 abstracts en double, dont 339 NA. 

In [41]:
data_brevets_NA = data_brevets_clean[
    data_brevets_clean["abstract"] == "NA"
]
data_brevets_NA

doc-number kind country  status publication_country  \
26036    3064421   A1      FR  PUBDEM                  FR   
26188    3064439   A1      FR  PUBDEM                  FR   
30080    3068197   A1      FR     NEW                  FR   
30673    3068791   A1      FR     NEW                  FR   
31322    3069440   A1      FR     NEW                  FR   
...          ...  ...     ...     ...                 ...   
50646    3088592   A1      FR     NEW                  FR   
50648    3088744   A1      FR     NEW                  FR   
50654    3088523   A1      FR     NEW                  FR   
50670    3088612   A1      FR     NEW                  FR   
50708    3088710   A1      FR     NEW                  FR   

      publication_doc-number publication_date publication_bopinum  \
26036                3064421         20180928             2018-39   
26188                3064439         20180928             2018-39   
30080                3068197         20181228             2018-52   
30673                3068791         20190111             2019-02   
31322                3069440         20190201             2019-05   
...                      ...              ...                 ...   
50646                3088592         20200522             2020-21   
50648                3088744         20200522             2020-21   
50654                3088523         20200522             2020-21   
50670                3088612         20200522             2020-21   
50708                3088710         20200522             2020-21   

      publication_nature                                    invention-title  \
26036             Brevet  ROTOR POUR MOTEUR OU GENERATEUR ELECTROMAGNETI...   
26188             Brevet  DISPOSITIF DE RETROVISION VIDEO POUR VEHICULE ...   
30080  Demande française  CARTOUCHE DE CARACTERISATION ELECTRONIQUE DES ...   
30673  Demande française  ANALYSE APPROFONDIE DE L'ORTEQUE DE VERRE OPHT...   
31322  Demande française                   RENOVATEUR UNIVERSEL DE KERATINE   
...                  ...                                                ...   
50646  Demande française  DISPOSITIF DE SECURITE POUR CIRCUIT-ELECTRIQUE...   
50648  Demande française  DISPOSITIF DE DETECTION DE LA PRESENCE DE PERS...   
50654  Demande française             Chaussures à semelles interchangeables   
50670  Demande française           Dispositif de récupération d'un véhicule   
50708  Demande française      ECHANGEUR DE CHALEUR POUR VEHICULE AUTOMOBILE   

              applicant_1_orgname                applicant_1_address-1  \
26036             WHYLOT SAS None        PARC D'ACTIVITES QUERCYPOLE 2   
26188                    DAV None  76 RUE AUGUSTE PERRET - ZI EUROPARC   
30080             SWISSLOG FRANCE                     28 QUAI GALLIENI   
30673            GIROD DENIS PAUL           71 BOULEVARD DE STRASBOURG   
31322  DUPIRE PIERRE JULES JOSEPH               66 RUE VAN DER MEERSCH   
...                           ...                                  ...   
50646                  LIVBAG SAS            2 RUE VILLARET DE JOYEUSE   
50648        INNOVATIVE RESOURCES                  60 ROUTE DU CHATEAU   
50654          OUABBAS ABDERRAHIM                   13 RUE DE LA BUTTE   
50670                 Naval Group          40-42 rue du Docteur Finlay   
50708   VALEO SYSTEMES THERMIQUES    8 RUE LOUIS LORMAND,  LA VERRIERE   

            applicant_1_city applicant_1_postcode applicant_1_country  \
26036                 CAMBES                46100                  FR   
26188          CRETEIL CEDEX                94046                  FR   
30080         SURESNES CEDEX                92156                  FR   
30673               LE HAVRE                76600                  FR   
31322              TOURCOING                59200                  FR   
...                      ...                  ...                 ...   
50646                  PARIS                75017                  FR   
50648              GRI

- **Gestion des NA** : En observant la base de données, on constate que ces abstract vides sont parfois associés à des demandes incomplètes, et parfois le reste de la demande semble complète. Après quelques recherches sur internet, il nous semble que l'explication la plus probable est que ces demandes soient déposées sans leur abstract, qui est fourni plus tard à l'INPI (l'abstract étant obligatoire pour la demande). Pour se concentrer sur les demandes complètes uniquement, nous décidons donc de supprimer ces observations, qui restent en faible nombre par rapport à nos plus de 100,000 observations.

In [49]:
# Drop des NA
data_brevets_clean = data_brevets_clean[
    data_brevets_clean["abstract"] != "NA"
]

# Vérification 
nb_na = (data_brevets_clean["abstract"] == "NA").sum()
print("Nombre de NA après le drop :", nb_na)

Nombre de NA après le drop : 0


- **Gestion des abstracts en double** : 

In [62]:
# compte des abstracts strictement identiques
duplicates_exact = data_brevets_clean[data_brevets_clean.duplicated(subset="abstract", keep=False)]
print("Nombre d'observations avec des doublons :", len(duplicates_exact))

Nombre d'observations avec des doublons : 2691


On commence par parcourir les observations avec des doublons pour se donner une idée de la raison de l'existence de ces doublons.

In [61]:
duplicates_exact.sort_values("abstract")

doc-number kind country  status publication_country  \
14977    3053138   A1      FR  PUBDEM                  FR   
74543    3112481   A1      FR     NEW                  FR   
65623    3103688   A1      FR     NEW                  FR   
65661    3103686   A1      FR     NEW                  FR   
59044    3097313   A1      FR     NEW                  FR   
...          ...  ...     ...     ...                 ...   
23601    3061660   A1      FR  PUBDEM                  FR   
9767     3048051   A1      FR  PUBDEM                  FR   
9851     3048052   A1      FR  PUBDEM                  FR   
16838    3055061   A1      FR  PUBDEM                  FR   
16884    3055060   A1      FR  PUBDEM                  FR   

      publication_doc-number publication_date publication_bopinum  \
14977                3053138       2017-12-29             2017-52   
74543                3112481       2022-01-21             2022-03   
65623                3103688       2021-06-04             2021-22   
65661                3103686       2021-06-04             2021-22   
59044                3097313       2020-12-18             2020-51   
...                      ...              ...                 ...   
23601                3061660       2018-07-13             2018-28   
9767                 3048051       2017-08-25             2017-34   
9851                 3048052       2017-08-25             2017-34   
16838                3055061       2018-02-16             2018-07   
16884                3055060       2018-02-16             2018-07   

      publication_nature                                    invention-title  \
14977             Brevet  PROCEDE D'EDITION AUTOMATISEE D'UN BLOC D'UN T...   
74543  Demande française  COMPOSITION POUR TEINDRE DES FIBRES DE KÉRATIN...   
65623  Demande française  Dispositif destiné à assurer la connexion de s...   
65661  Demande française  Dispositif destiné à assurer la connexion de s...   
59044  Demande française  Évaporateur d’un fluide de travail pour une ce...   
...                  ...                                                ...   
23601             Brevet  COMPOSITIONS SOUS FORME D'UNE SOLUTION AQUEUSE...   
9767              Brevet  DISPOSITIF DE RACCORDEMENT, PROCEDE ET OUTIL P...   
9851              Brevet  DISPOSITIF DE RACCORDEMENT, PROCEDE ET OUTIL P...   
16838             Brevet                         ECRAN MOBILE POUR VEHICULE   
16884             Brevet                         ECRAN MOBILE POUR VEHICULE   

                                 applicant_1_orgname  \
14977                                LUCKY CART None   
74543                                        L'OREAL   
65623                                  READYNOVATION   
65661                                  READYNOVATION   
59044                                 NAVAL ENERGIES   
...                                              ...   
23601                                    ADOCIA None   
9767   PARKER HANNIFIN MANUFACTURING FRANCE SAS None   
9851   PARKER HANNIFIN MANUFACTURING FRANCE SAS None   
16838            PEUGEOT CITROEN AUTOMOBILES SA None   
16884            PEUGEOT CITROEN AUTOMOBILES SA None   

                                   applicant_1_address-1     applicant_1_city  \
14977                                 159 RUE MONTMARTRE                PARIS   
74543                                     14, rue Royale                PARIS   
65623        9 rue Volta Chez Monsieur Geoffroy GLORIEUX                PARIS   
65661                                        9 rue Volta                PARIS   
59044                        40-42 rue du Docteur Finlay                PARIS   
...                                                  ...                  ...   
23601                              115 AVENUE LACASSAGNE                 LYON   
9767   17 RUE DES BUCHILLONS\nZI DU MONT BLANC - VILL...            ANNEMASSE   
9851   17 RUE DES BUCHILLONS\nZI DU MONT BLANC - VILL...            ANNEMASSE   
16838       

**1. Cas des doublons parfaits :**

On veut d'abord vérifier si il existe des observations qui ont exactement les mêmes valeurs pour toutes les variables sauf pour le numéro de dossier.

In [ ]:
# Vérification existance doublon parfait
cols_comparaison = [
    c for c in data_brevets_clean.columns
    if c not in ["doc_number", "publication_doc_number"]
]


n_duplicates = data_brevets_clean.duplicated(
    subset=cols_comparaison
).sum()
print(n_duplicates)

Il n'y a donc aucun doublon parfait ! Toutes les demandes avec le même abstract ...

**2. Cas des doublons avec des dates différentes :**

Nous émettons l'hypothèse que les observations avec des NA en double correspondent à la même demande de brevet déposée plusieurs fois afin d'ajouter une correction d'une version à une autre. Ainsi, nous choisissons de considérer que la demande la plus récente entre les doublons est la demande la plus aboutie, et donc la seule que nous garderons. 

In [ ]:
# Garder la demande la plus ancienne parmi les doublons

## Passer publication_date au format date
data_brevets_clean["publication_date"] = pd.to_datetime(
    data_brevets_clean["publication_date"],
    format="%Y%m%d",
    errors="coerce"
)
## Trouver la date la plus ancienne par abstract
max_date_per_abstract = (
    duplicates_exact
    .groupby("abstract")["publication_date"]
    .transform("max")
)

## Garder les doublons les plus récents
duplicates_keep = duplicates_exact[
    duplicates_exact["publication_date"] == max_date_per_abstract
]

## Reconstruire la base
data_brevets_clean2 = pd.concat(
    [
        data_brevets_clean[~data_brevets_clean.index.isin(duplicates_exact.index)],
        duplicates_keep
    ],
    axis=0
).sort_index()

## Compter le nouveau nombre de brevets
duplicates_exact2 = data_brevets_clean2[data_brevets_clean2.duplicated(subset="abstract", keep=False)]
print("Nouveau nombre d'observations avec des doublons :", len(duplicates_exact2))


Nombre d'observations avec des doublons : 1872


0


In [44]:
data_brevets_clean.groupby("abstract").filter(lambda x: len(x) > 1)

doc-number kind country  status publication_country  \
48        3038286   A1      FR  PUBDEM                  FR   
76        3038249   A1      FR  PUBDEM                  FR   
84        3038325   A1      FR  PUBDEM                  FR   
131       3038279   A1      FR  PUBDEM                  FR   
160       3038242   A1      FR  PUBDEM                  FR   
...           ...  ...     ...     ...                 ...   
112223    3150324   A1      FR     NEW                  FR   
112224    3150325   A1      FR     NEW                  FR   
112225    3150326   A1      FR     NEW                  FR   
112288    3150389   A1      FR     NEW                  FR   
112289    3150390   A1      FR     NEW                  FR   

       publication_doc-number publication_date publication_bopinum  \
48                    3038286         20170106             2017-01   
76                    3038249         20170106             2017-01   
84                    3038325         20170106             2017-01   
131                   3038279         20170106             2017-01   
160                   3038242         20170106             2017-01   
...                       ...              ...                 ...   
112223                3150324         20241227             2024-52   
112224                3150325         20241227             2024-52   
112225                3150326         20241227             2024-52   
112288                3150389         20241227             2024-52   
112289                3150390         20241227             2024-52   

       publication_nature                                    invention-title  \
48                 Brevet  ENSEMBLE COMPRENANT UN GARDE-BOUE POUR VEHICUL...   
76                 Brevet  VITRAGE CHAUFFANT A FEUILLE DE VERRE EXTERIEUR...   
84                 Brevet  PROCEDE DE CRYOCONSERVATION DE CELLULES A VISE...   
131                Brevet  PROCEDE DE CONDUITE AUTOMATIQUE POUR L'INSERTI...   
160                Brevet  ALLIAGE D'ALUMINIUM POUR SOUDAGE PAR LASER SAN...   
...                   ...                                                ...   
112223  Demande française  Procédé de traitement informatique de données ...   
112224  Demande française  Procédé de traitement informatique de données ...   
112225  Demande française  Procédé de traitement informatique de données ...   
112288  Demande française  Dispositif de refroidissement configuré pour r...   
112289  Demande française  Dispositif de refroidissement configuré pour r...   

                                      applicant_1_orgname  \
48                    PEUGEOT CITROEN AUTOMOBILES SA None   
76                         SAINT-GOBAIN GLASS FRANCE None   
84      LABORATOIRE FRANCAIS DU FRACTIONNEMENT ET DES ...   
131     COMMISSARIAT A L'ENERGIE ATOMIQUE ET AUX ENERG...   
160                         CONSTELLIUM NEUF-BRISACH None   
...                                                   ...   
112223                                       Rayon Thomas   
112224                                       Rayon Thomas   
112225                                       Rayon Thomas   
112288                          VALEO SYSTEMES THERMIQUES   
112289                          VALEO SYSTEMES THERMIQUES   

                                    applicant_1_address-1  \
48                                          ROUTE DE GISY   
76                                     18 AVENUE D'ALSACE   
84                   ZA DE COURTABOEUF 3 AV DES TROPIQUES   
131              BATIMENT "LE PONANT D", \n25 RUE LEBLANC   
160                               ZIP RHENANE NORD\nRD 52   
...                                                   ...   
112223  Entrée Boutique Coup d'oeil 6 Rue Joseph Marie...   
112224  Entrée Boutique Coup d'oeil 6 Rue Joseph Marie...   
112225  Entrée Boutique Coup d'oeil 6 Rue Joseph Marie...   
112288                    8 rue Louis Lormand La Verrière   
112289                    8 rue Louis Lormand La Verrièr

In [45]:
data_brevets_clean["abstract"].nunique()


105885

### Tri des variables pertinentes

In [17]:
data_brevets_light = data_brevets[['publication_date', 'invention-title', 'classification_1_text', 'abstract']]
data_brevets_clean.head(5)


doc-number kind country  status publication_country publication_doc-number  \
0    3038352   A1      FR  PUBDEM                  FR                3038352   
1    3038278   A1      FR  PUBDEM                  FR                3038278   
2    3038409   A1      FR  PUBDEM                  FR                3038409   
3    3038425   A1      FR  PUBDEM                  FR                3038425   
4    3038316   A1      FR  PUBDEM                  FR                3038316   

  publication_date publication_bopinum publication_nature  \
0         20170106             2017-01             Brevet   
1         20170106             2017-01             Brevet   
2         20170106             2017-01             Brevet   
3         20170106             2017-01             Brevet   
4         20170106             2017-01             Brevet   

                                     invention-title  ... citation_3_country  \
0  ENSEMBLE POUR PANNEAU DE PORTE ET PANNEAU ASSOCIE  ...                 NA   
1  VEHICULE HYBRIDE COMPRENANT UNE SUPPRESSION DE...  ...                 NA   
2  PROCEDE DE CONCEPTION ASSISTEE PAR ORDINATEUR ...  ...                 NA   
3  DOCUMENT ELECTRONIQUE TEL QU'UNE CARTE A PUCE ...  ...                 NA   
4  PROCEDE DE SYNTHESE EN CONTINU D'UN ELASTOMERE...  ...                 NA   

  citation_3_doc-number citation_3_date  year application_country  \
0                    NA              NA  2017                None   
1                    NA              NA  2017                None   
2                    NA              NA  2017                None   
3                    NA              NA  2017                None   
4                    NA              NA  2017                None   

  application_doc-number application_date last-fee-payement next-fee-payement  \
0                   None             None              None              None   
1                   None             None              None              None   
2                   None             None              None              None   
3                   None             None              None              None   
4                   None             None              None              None   

  date-search-completed  
0                  None  
1                  None  
2                  None  
3                  None  
4                  None  

[5 rows x 102 columns]